Tornado AssertionError when starting salt-minion on deb/wheezy (develop)

I'm trying to the test my changes from #24422 using the latest develop branch.

```
./bootstrap-salt.sh -P -M -K -g https://github.com/bechtoldt/salt.git git refactor_es_modules
```

```
=> master1: Downloading/unpacking tornado>=4.0
==> master1:   Running setup.py egg_info for package tornado
==> master1:
==> master1:     no previously-included directories found matching 'docs/build'
==> master1:     warning: no files found matching 'tornado/test/README'
==> master1: Downloading/unpacking backports.ssl-match-hostname (from tornado>=4.0)
==> master1:   Downloading backports.ssl_match_hostname-3.4.0.2.tar.gz
==> master1:   Running setup.py egg_info for package backports.ssl-match-hostname
==> master1:
==> master1: Downloading/unpacking certifi (from tornado>=4.0)
==> master1:   Running setup.py egg_info for package certifi
==> master1:
==> master1: Installing collected packages: tornado, backports.ssl-match-hostname, certifi
==> master1:   Running setup.py install for tornado
==> master1:     building 'tornado.speedups' extension
==> master1:     gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c tornado/speedups.c -o build/temp.linux-x86_64-2.7/tornado/speedups.o
==> master1:     tornado/speedups.c:49:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
==> master1:     gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/tornado/speedups.o -o build/lib.linux-x86_64-2.7/tornado/speedups.so
==> master1:
==> master1:     no previously-included directories found matching 'docs/build'
==> master1:     warning: no files found matching 'tornado/test/README'
==> master1:   Running setup.py install for backports.ssl-match-hostname
==> master1:
==> master1:   Running setup.py install for certifi
==> master1:
==> master1: Successfully installed tornado backports.ssl-match-hostname certifi
```

```
# salt-minion --versions
Salt Version:
           Salt: 2015.5.2-5695-g2c2cc20

Dependency Versions:
         Jinja2: 2.6
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 13.1.0
         Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
           RAET: Not Installed
            ZMQ: 3.2.3
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.1.10
       pycrypto: 2.6

System Versions:
           dist: debian 7.8
        machine: x86_64
        release: 3.2.0-4-amd64
         system: debian 7.8
```

```

# pip freeze
Warning: cannot find svn location for distribute==0.6.24dev-r0
Jinja2==2.6
M2Crypto==0.21.1
MarkupSafe==0.15
PyYAML==3.10
argparse==1.2.1
backports.ssl-match-hostname==3.4.0.2
certifi==2015.04.28
chardet==2.0.1
## FIXME: could not find svn URL in dependency_links for this package:
distribute==0.6.24dev-r0
msgpack-python==0.1.10
pycrypto==2.6
python-apt==0.8.8.2
pyzmq==13.1.0
requests==2.0.0
salt==2015.5.2-5695-g2c2cc20
six==1.1.0
tornado==4.2
urllib3==1.7.1
wsgiref==0.1.2
```

```
2015-06-05 11:15:14,072 [salt.scripts                             ][ERROR   ][25853] Minion failed to start:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 81, in minion_process
    minion.start()
  File "/usr/lib/python2.7/dist-packages/salt/cli/daemons.py", line 269, in start
    self.minion.tune_in()
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1593, in tune_in
    self.sync_connect_master()
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 589, in sync_connect_master
    raise six.reraise(*future_exception)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 879, in run
    yielded = self.gen.send(value)
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 597, in connect_master
    yield self._post_master_init(master)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 242, in wrapper
    Runner(result, future, yielded)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 817, in __init__
    if self.handle_yield(first_yielded):
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 956, in handle_yield
    self.future = convert_yielded(yielded)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 1022, in convert_yielded
    return multi_future(yielded)
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 643, in multi_future
    assert all(is_future(i) for i in children)
AssertionError
2015-06-05 11:15:14,073 [salt.scripts                             ][WARNING ][25853] ** Restarting minion **
```

--- a/salt/pillar/__init__.py
+++ b/salt/pillar/__init__.py
@@ -68,8 +68,8 @@ def get_async_pillar(opts, grains, id_, saltenv=None, ext=None, env=None, funcs=
         saltenv = env
     ptype = {
         'remote': AsyncRemotePillar,
-        #'local': AsyncPillar  # TODO: implement
-    }.get(opts['file_client'], Pillar)
+        'local': AsyncPillar,
+    }.get(opts['file_client'], AsyncPillar)
     return ptype(opts, grains, id_, saltenv, ext, functions=funcs,
                  pillar=pillar, pillarenv=pillarenv)
 
@@ -688,3 +688,12 @@ class Pillar(object):
                 log.critical('Pillar render error: {0}'.format(error))
             pillar['_errors'] = errors
         return pillar
+
+
+# TODO: actually migrate from Pillar to AsyncPillar to allow for futures in
+# ext_pillar etc.
+class AsyncPillar(Pillar):
+    @tornado.gen.coroutine
+    def compile_pillar(self, ext=True, pillar_dirs=None):
+        ret = super(AsyncPillar, self).compile_pillar(ext=ext, pillar_dirs=pillar_dirs)
+        raise tornado.gen.Return(ret)

